lib/Dialects/VectorExt/Transform/VectorWarpUtils.cpp

```diff
@@ -0,0 +1,264 @@
+//===- VectorWarpUtils.cpp - Utilities vector warp ops --------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Dialects/VectorExt/VectorExtOps.h"
+#include "Dialects/VectorExt/VectorExtWarpUtils.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Vector/VectorOps.h"
+#include "mlir/IR/AffineMap.h"
+#include "mlir/IR/BlockAndValueMapping.h"
+#include "mlir/Support/LogicalResult.h"
+
+using namespace mlir;
+using namespace mlir::vector;
+using namespace mlir::vector_ext;
+
+// Clones `op` into a new operations that takes `operands` and returns
+// `resultTypes`.
+static Operation *cloneOpWithOperandsAndTypes(OpBuilder &builder, Location loc,
+                                              Operation *op,
+                                              ArrayRef<Value> operands,
+                                              ArrayRef<Type> resultTypes) {
+  OperationState res(loc, op->getName().getStringRef(), operands, resultTypes,
+                     op->getAttrs());
+  return builder.createOperation(res);
+}
+
+/// Helper to create a new WarpSingleLaneOp regions with different signature.
+static WarpSingleLaneOp
+moveRegionToNewWarpOpAndOverrideReturns(OpBuilder &b, WarpSingleLaneOp warpOp,
```

Comment:
nit: override -> replace.
Override has specific meanings that don't seem to match here.